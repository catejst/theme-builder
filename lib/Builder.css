html {
  box-sizing: border-box;
}
*, *:before, *:after {
  box-sizing: inherit;
}

.keen-theme-builder {
  padding: 1rem;
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: repeat(2, calc(50% - 0.5rem));
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  background-color: #F3F4F7;

  .builder {

    &__config,
    &__mockup {
      padding: 1rem;
      background-color: #fff;
      box-sizing: border-box;
    }

    &__select {
      display: block;
      margin: 0.5rem 0;
    }

    &__editor {
      padding: 10px;
      border: 1px solid #d6d6d6;
      background-color: #fff;
    }
  }

  .color-picker {
    position: relative;
    margin-bottom: 1rem;
    display: flex;
    flex-direction: column;

    &__label {
      margin-bottom: 0.5rem;
      font-weight: bold;
    }

    &__wrapper {
      padding: 0.25rem 0.5rem;
      display: inline-flex;
      align-items: center;
      justify-content: space-between;
      border-radius: 3px;
      border: 1px solid lightgray;
    }

    &__input {
      border: none;
      outline: none;
      background: transparent;
    }

    &__trigger {
      width: 15px;
      height: 15px;
      padding: 2px;
      border: 1px solid lightgray;
      border-radius: 50%;
      cursor: pointer;
      transition: all 300ms ease;

      &:hover {
        transform: scale(1.15);
      }
    }

    &__popover {
      position: absolute;
      top: 0;
      left: 100%;
      z-index: 2;
    }

    &__cover {
      position: fixed;
      top: 0;
      bottom: 0;
      left: 0;
      right: 0;
      z-index: -1;
    }
  }

  .color-palette {

    &__wrapper {
      display: inline-flex;
      flex-wrap: wrap;
    }

    &-item {
      position: relative;

      &:hover {
        .color-palette-button {
          display: block;
        }
      }
    }

    &-button {
      display: none;
      position: absolute;
      z-index: 100;
      bottom: 100%;
      left: 50%;
      transform: translateX(-50%);
      cursor: pointer;
    }

    &__trigger {
      width: 20px;
      height: 20px;
      padding: 2px;
      border: 2px solid lightgray;
      border-radius: 50%;
      cursor: pointer;
      transition: all 300ms ease;

      &:hover {
        transform: scale(1.15);
      }
    }
  }

  .keen-dataviz {
    height: 50vh;
    padding: 1rem;
    background-color: #fff;
    border: 1px solid #d6d6d6;
  }

  .react-tabs {

    &__tab {
      &:focus {
        box-shadow: none;

        &:after {
          content: none;
        }
      }

      &-list {
        margin: 0;
        border-color: #d6d6d6;
      }
  
      &-panel {
        padding: 1rem;
        background-color: #F3F4F7;
        border-color: #d6d6d6;
        border-style: solid;
        border-width: 0 1px 1px 1px;
      }
  
      &--selected {
        border-color: #d6d6d6;
        background-color: #F3F4F7;
      }
    }
  }
}