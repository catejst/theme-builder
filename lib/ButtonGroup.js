import React from 'react';
import PropTypes from 'prop-types';

export const ButtonGroup = (props) => {
  const { items, singleChoice, groupName, activeItem, onChange } = props;
  const buttons = items.map((item) => {
    return singleChoice ?
    <label className={`button-group__label ${item.value === activeItem && 'active'}`} key={item.value}>
      {item.label}
      <input type="radio" value={item.value} checked={item.value === activeItem} name={groupName} onChange={onChange} className="button-group__input"></input>
    </label> :
    <label className={`button-group__label ${activeItem[item.value] && 'active'}`} key={item.value}>
      {item.label}
      <input type="checkbox" value={item.value} onChange={onChange} className="button-group__input"></input>
    </label>
  });
  
  return (
    <div className="button-group">
      {buttons}
    </div>
  );
};
ButtonGroup.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    value: PropTypes.string,
  })).isRequired,
  singleChoice: PropTypes.bool.isRequired,
  groupName: PropTypes.string,
  activeItem: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
  onChange: PropTypes.func.isRequired,
};

ButtonGroup.defaultProps = {
  items: [],
  singleChoice: false,
  activeItem: [],
  onChange: () => {},
}