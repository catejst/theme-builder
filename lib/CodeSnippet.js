import React from 'react';
import PropTypes from 'prop-types';
import { stripIndent } from 'common-tags';
import Lowlight from 'react-lowlight';
import css from 'highlight.js/lib/languages/css';
import js from 'highlight.js/lib/languages/javascript';
import 'highlight.js/styles/github-gist.css';
import { getCSSRules } from './utils/helpers';

Lowlight.registerLanguage('css', css);
Lowlight.registerLanguage('js', js);

const placeholder = document.createElement('li');
placeholder.className = 'placeholder'; 

const CodeSnippet = ({ type, data, ...props }) => {
  const getJSCode = () => {
    const colors = data.colors || [];
    return stripIndent`
      const chart = new KeenDataviz({
        container: '#some_container', // required
        colors: ${JSON.stringify(colors)}
      });
    `;
  }

  const getCSSCode = () => {
    const {
      chartBackground,
      chartBorder,
      chartFont,
      titleTextAlign,
      titleColor,
      titleFontFamily,
      titleFontSize,
      titleFontBold,
      titleFontItalic,
      axisYLabelFont,
      axisYLabelFontSize,
      axisYLabelFontBold,
      axisYLabelFontItalic,
      axisYLabelColor,
      axisXLabelFont,
      axisXLabelFontSize,
      axisXLabelFontBold,
      axisXLabelFontItalic,
      axisXLabelColor,
      legendFont,
      legendFontSize,
      legendFontBold,
      legendFontItalic,
      legendColor,
      gridLineShow,
      gridLineColor,
      subgridLineColor,
    } = getCSSRules(data);
    return stripIndent`
      .keen-theme-builder .keen-dataviz {
        ${chartBackground}
        ${chartBorder}
      }
      .keen-dataviz,
      .c3 svg {
        ${chartFont}
      }
      .keen-dataviz .keen-dataviz-title {
        display: block;
        ${titleColor}
        ${titleTextAlign}
        ${titleFontFamily}
        ${titleFontSize}
        ${titleFontBold}
        ${titleFontItalic}
      }
      .keen-dataviz .text-label {}
      .keen-dataviz .text-main,
      .keen-dataviz .text-second,
      .keen-dataviz .c3-chart-arcs .c3-chart-arcs-gauge-max,
      .keen-dataviz .c3-chart-arcs .c3-chart-arcs-gauge-min,
      .keen-dataviz .c3-chart-arc .c3-gauge-value,
      .keen-dataviz .c3-chart-arc text {}
      .keen-dataviz .c3-tooltip,
      .keen-dataviz .c3-tooltip-container th {}
      .keen-dataviz .c3-axis.c3-axis-y {
        ${axisYLabelFont}
        ${axisYLabelFontSize}
        ${axisYLabelFontBold}
        ${axisYLabelFontItalic}
      }
      .keen-dataviz .c3-axis.c3-axis-y .tick text {
        ${axisYLabelColor}
      }
      .keen-dataviz .c3-axis.c3-axis-x {
        ${axisXLabelFont}
        ${axisXLabelFontSize}
        ${axisXLabelFontBold}
        ${axisXLabelFontItalic}
      }
      .keen-dataviz .c3-axis.c3-axis-x .tick text {
        ${axisXLabelColor}
      }
      .keen-dataviz .keen-c3-legend .legend-item-text {
        ${legendFont}
        ${legendFontSize}
        ${legendFontBold}
        ${legendFontItalic}
        ${legendColor}
      }
      .keen-dataviz .c3-grid line,
      .keen-dataviz .chart-lines {
        ${gridLineColor}
      }
      .keen-theme-builder .keen-dataviz .c3-ygrid {
        ${gridLineShow}
      }
    `; 
  }

  const handleClickToCopy = (event, value) => {
    event.preventDefault();

    const placeholder = document.createElement('textarea');
    placeholder.value = value;
    placeholder.classList = 'copy-to-clipboard';
    
    document.body.appendChild(placeholder);
    placeholder.select();
    document.execCommand('copy');
    
    document.body.removeChild(placeholder);
  }

  const snippetData = type === 'js' ? getJSCode() : getCSSCode();

  const renderCopyButton = (value) => {
    return (
      <button className="builder__btn-copy btn" onClick={(e) => handleClickToCopy(e, value)}>Copy to Clipboard</button>
    )
  };

  return (
    <>
      {renderCopyButton(snippetData)}
      <Lowlight language={type} value={snippetData} className="builder__editor" />
    </>
  )
};

export default CodeSnippet;

CodeSnippet.defaultProps = {
  type: "js",
  data: {}
}

CodeSnippet.propTypes = {
  type: PropTypes.oneOf(['js', 'css']).isRequired,
  data: PropTypes.object.isRequired
}