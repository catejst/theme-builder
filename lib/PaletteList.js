import React, {
  useState,
  useEffect,
  useRef,
} from 'react';
import {
  sortableContainer,
  sortableElement,
  sortableHandle
} from 'react-sortable-hoc';
import arrayMove from 'array-move';
import {
  SketchPicker
} from 'react-color';
import isEqual from 'lodash.isequal';
import {  defaultColor } from './config';

const DragHandle = sortableHandle(() => <div style={{ color: 'rgba(255, 255, 255, 0.4)'}}>&harr;</div>);

const Item = sortableElement(props => {
  const { value, index, removeItem, changeColor } = props;
  const [isPickerOpen, setPicker] = useState(false);
  const openPicker = () => {
    setPicker(true);
  };
  const closePicker = () => {
    setPicker(false);
  };
  return (
    <div className="color-palette__item">
      <button className="color-palette__button btn-remove"
        type="button"
        onClick={() => removeItem(index)}
        style={{
          position: 'absolute',
          top: 0,
          right: 0,
          zIndex: 1,
          color: 'red',
          padding: '3px 6px',
          backgroundColor: 'white',
          border: 'none',
          display: 'none',
        }}>
        &times;
      </button>
      <button className="color-palette__button btn-edit"
        type="button"
        onClick={openPicker}
        style={{
          position: 'absolute',
          top: '100%',
          left: 0,
          zIndex: 1,
          width: '100%',
          display: 'none',
        }}>
        Edit
      </button>
      <div className="color-palette__trigger" style={{
        backgroundColor: value,
        width: '50px',
        height: '50px',
        padding: '2px',
        border: '1px solid #ccc',
        display: 'inline-flex',
        alignItems: 'center',
        justifyContent: 'center',
        position: 'relative',
        }}>
        <DragHandle/>
      </div>
      { isPickerOpen
        ? 
        <div className="color-picker__popover">
          <div className="color-picker__cover" onClick={closePicker}></div>
          <SketchPicker
            className='color-picker__picker'
            color={value}
            onChangeComplete={ event => changeColor(event.hex, index) }
            disableAlpha={true} />
        </div>              
        : null }
    </div>
  );
});

const Container = sortableContainer(({children}) => {
  return <div className="color-palette__list">{children}</div>;
});

export const List = (props) => {
  const { colors, onChange } = props;
  const [ stateColors, setStateColors ] = useState([...colors]);
  const prevColorsRef = useRef();
  const prevColors = prevColorsRef.current;
  useEffect(() => {
    prevColorsRef.current = stateColors;
    if (!isEqual(stateColors, colors)) {
      onChange(stateColors);
    }
    if (!isEqual(prevColors, colors)) {
      setStateColors(colors);
    }
  });

  const onSortEnd = ({oldIndex, newIndex}) => {
    const sortedColors = arrayMove(stateColors, oldIndex, newIndex);
    setStateColors(sortedColors);
  };

  const addItem = () => {
    const addedColors = [...stateColors, defaultColor];
    setStateColors(addedColors);
  }

  const removeItem = (index) => {
    const removedColors = [...stateColors];
    removedColors.splice(index,1);
    setStateColors(removedColors);
  };

  const changeColor = (color, index) => {
    const changedColors = [...stateColors];
    changedColors[index] = color;
    setStateColors(changedColors);
  };

  return (
    <>
    <Container onSortEnd={onSortEnd} useDragHandle axis='xy'>
      {stateColors.map((value, index) => (
        <Item key={`color-${index}`} index={index} value={value} removeItem={removeItem} changeColor={changeColor} />
      ))}
      <button className="color-palette__button btn-list btn" type="button" onClick={addItem}>Add</button>
    </Container>
    </>
  );
};