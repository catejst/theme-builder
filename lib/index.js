import Builder from './components/Builder';
import { getCSSCode } from './utils/helpers';

export default Builder;
export const getStyles = (data, id) => getCSSCode(data, id);
