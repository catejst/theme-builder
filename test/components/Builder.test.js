import React from 'react'
import { shallow } from 'enzyme';
import Builder from '../../lib/Builder';
import { chartColorPalette, defaultColor } from '../../lib/config';

describe('<Builder />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Builder />);
  });

  it('Builder component renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  
  it('should render list of available chart types', () => {
    expect(wrapper.find('.chart-types').length).toBe(1);
  })
  
  it('should render list of predefined color palettes', () => {
    expect(wrapper.find('.predefined-palette').length).toBe(1);
  })
  
  it('should open color picker on trigger click', () => {
    wrapper.find('.color-picker__trigger').first().simulate('click');
    expect(wrapper.find('.color-picker__picker').length).toBe(1);
  });
  
  it('should close color picker once click outside the picker', () => {
    wrapper.find('.color-picker__trigger').first().simulate('click');
    wrapper.find('.color-picker__cover').simulate('click');
    expect(wrapper.find('.color-picker__picker').length).toBe(0);
  });

  it('should render tabs', () => {
    expect(wrapper.find('.react-tabs__tab').length).toBe(3);
  })

  it('should render default color value for title color input', () => {
    expect(wrapper.find('.color-picker__input').first().props().value).toEqual(defaultColor);
  });

  it('should call copy to clipboard method when click on the button', () => {
    const copyToClipboard = jest.fn();
    wrapper.instance().handleClickToCopy = copyToClipboard;
    wrapper.find('.builder__btn-copy').first().simulate('click');
    expect(copyToClipboard).toBeCalled();
  });

  describe('props given', () => {
    let wrapper;
    const onChange = jest.fn();
    const palette = { value: 'modern', label: 'Modern' };

    beforeEach(() => {
      wrapper = shallow(<Builder chartPalette={palette} onChange={onChange} />);
    });

    it('should update state with chosen color palette', () => {
      expect(wrapper.state().colors).toEqual(chartColorPalette['modern'].colors);
    });

    it('should not render chart types once Dashboard Builder is active', () => {
      wrapper.setProps({ isDashboardBuilderActive: true });
      expect(wrapper.find('.chart-types').length).toBe(0);
    })

    it('should not render tabs once Dashboard Builder is active', () => {
      wrapper.setProps({ isDashboardBuilderActive: true });
      expect(wrapper.find('.react-tabs__tab').length).toBe(0);
    });
  })
});
