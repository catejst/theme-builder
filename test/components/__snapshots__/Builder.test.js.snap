// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Builder /> Builder component renders correctly 1`] = `"<div class=\\"keen-theme-builder\\"><section class=\\"builder__config\\"><h2 class=\\"builder__header\\">Chart type</h2><label>Available chart types</label><div class=\\"chart-types css-2b097c-container\\"><div class=\\" css-yk16xz-control\\"><div class=\\" css-1hwfws3\\"><div class=\\" css-1uccc91-singleValue\\">Default</div><div class=\\"css-1g6gooi\\"><div class=\\"\\" style=\\"display:inline-block\\"><input type=\\"text\\" autoCapitalize=\\"none\\" autoComplete=\\"off\\" autoCorrect=\\"off\\" id=\\"react-select-2-input\\" spellcheck=\\"false\\" tabindex=\\"0\\" value=\\"\\" aria-autocomplete=\\"list\\" style=\\"box-sizing:content-box;width:1px;label:input;background:0;border:0;font-size:inherit;opacity:1;outline:0;padding:0;color:inherit\\"/><div style=\\"position:absolute;top:0;left:0;visibility:hidden;height:0;overflow:scroll;white-space:pre\\"></div></div></div></div><div class=\\" css-1wy0on6\\"><span class=\\" css-1okebmr-indicatorSeparator\\"></span><div aria-hidden=\\"true\\" class=\\" css-tlfecz-indicatorContainer\\"><svg height=\\"20\\" width=\\"20\\" viewBox=\\"0 0 20 20\\" aria-hidden=\\"true\\" focusable=\\"false\\" class=\\"css-19bqh2r\\"><path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\"></path></svg></div></div></div></div><h2 class=\\"builder__header\\">Font</h2><div class=\\"chart-fonts css-2b097c-container\\"><div class=\\" css-yk16xz-control\\"><div class=\\" css-1hwfws3\\"><div class=\\" css-1uccc91-singleValue\\">Default</div><div class=\\"css-1g6gooi\\"><div class=\\"\\" style=\\"display:inline-block\\"><input type=\\"text\\" autoCapitalize=\\"none\\" autoComplete=\\"off\\" autoCorrect=\\"off\\" id=\\"react-select-3-input\\" spellcheck=\\"false\\" tabindex=\\"0\\" value=\\"\\" aria-autocomplete=\\"list\\" style=\\"box-sizing:content-box;width:1px;label:input;background:0;border:0;font-size:inherit;opacity:1;outline:0;padding:0;color:inherit\\"/><div style=\\"position:absolute;top:0;left:0;visibility:hidden;height:0;overflow:scroll;white-space:pre\\"></div></div></div></div><div class=\\" css-1wy0on6\\"><span class=\\" css-1okebmr-indicatorSeparator\\"></span><div aria-hidden=\\"true\\" class=\\" css-tlfecz-indicatorContainer\\"><svg height=\\"20\\" width=\\"20\\" viewBox=\\"0 0 20 20\\" aria-hidden=\\"true\\" focusable=\\"false\\" class=\\"css-19bqh2r\\"><path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\"></path></svg></div></div></div></div><div class=\\"button-group\\"><label class=\\"button-group__label false\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"bold\\" class=\\"svg-inline--fa fa-bold fa-w-12 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 384 512\\"><path fill=\\"currentColor\\" d=\\"M333.49 238a122 122 0 0 0 27-65.21C367.87 96.49 308 32 233.42 32H34a16 16 0 0 0-16 16v48a16 16 0 0 0 16 16h31.87v288H34a16 16 0 0 0-16 16v48a16 16 0 0 0 16 16h209.32c70.8 0 134.14-51.75 141-122.4 4.74-48.45-16.39-92.06-50.83-119.6zM145.66 112h87.76a48 48 0 0 1 0 96h-87.76zm87.76 288h-87.76V288h87.76a56 56 0 0 1 0 112z\\"></path></svg><input type=\\"checkbox\\" value=\\"bold\\" class=\\"button-group__input\\"/></label><label class=\\"button-group__label false\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"italic\\" class=\\"svg-inline--fa fa-italic fa-w-10 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 320 512\\"><path fill=\\"currentColor\\" d=\\"M320 48v32a16 16 0 0 1-16 16h-62.76l-80 320H208a16 16 0 0 1 16 16v32a16 16 0 0 1-16 16H16a16 16 0 0 1-16-16v-32a16 16 0 0 1 16-16h62.76l80-320H112a16 16 0 0 1-16-16V48a16 16 0 0 1 16-16h192a16 16 0 0 1 16 16z\\"></path></svg><input type=\\"checkbox\\" value=\\"italic\\" class=\\"button-group__input\\"/></label></div><div class=\\"button-group\\"><label class=\\"button-group__label active\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"align-left\\" class=\\"svg-inline--fa fa-align-left fa-w-14 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 448 512\\"><path fill=\\"currentColor\\" d=\\"M12.83 352h262.34A12.82 12.82 0 0 0 288 339.17v-38.34A12.82 12.82 0 0 0 275.17 288H12.83A12.82 12.82 0 0 0 0 300.83v38.34A12.82 12.82 0 0 0 12.83 352zm0-256h262.34A12.82 12.82 0 0 0 288 83.17V44.83A12.82 12.82 0 0 0 275.17 32H12.83A12.82 12.82 0 0 0 0 44.83v38.34A12.82 12.82 0 0 0 12.83 96zM432 160H16a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0 256H16a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z\\"></path></svg><input type=\\"radio\\" value=\\"left\\" checked=\\"\\" name=\\"textAlignment\\" class=\\"button-group__input\\"/></label><label class=\\"button-group__label false\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"align-center\\" class=\\"svg-inline--fa fa-align-center fa-w-14 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 448 512\\"><path fill=\\"currentColor\\" d=\\"M432 160H16a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0 256H16a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zM108.1 96h231.81A12.09 12.09 0 0 0 352 83.9V44.09A12.09 12.09 0 0 0 339.91 32H108.1A12.09 12.09 0 0 0 96 44.09V83.9A12.1 12.1 0 0 0 108.1 96zm231.81 256A12.09 12.09 0 0 0 352 339.9v-39.81A12.09 12.09 0 0 0 339.91 288H108.1A12.09 12.09 0 0 0 96 300.09v39.81a12.1 12.1 0 0 0 12.1 12.1z\\"></path></svg><input type=\\"radio\\" value=\\"center\\" name=\\"textAlignment\\" class=\\"button-group__input\\"/></label><label class=\\"button-group__label false\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"align-right\\" class=\\"svg-inline--fa fa-align-right fa-w-14 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 448 512\\"><path fill=\\"currentColor\\" d=\\"M16 224h416a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16H16a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16zm416 192H16a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm3.17-384H172.83A12.82 12.82 0 0 0 160 44.83v38.34A12.82 12.82 0 0 0 172.83 96h262.34A12.82 12.82 0 0 0 448 83.17V44.83A12.82 12.82 0 0 0 435.17 32zm0 256H172.83A12.82 12.82 0 0 0 160 300.83v38.34A12.82 12.82 0 0 0 172.83 352h262.34A12.82 12.82 0 0 0 448 339.17v-38.34A12.82 12.82 0 0 0 435.17 288z\\"></path></svg><input type=\\"radio\\" value=\\"right\\" name=\\"textAlignment\\" class=\\"button-group__input\\"/></label></div><h2 class=\\"builder__header\\">Custom colors</h2><div class=\\"custom-color-picker\\"><div class=\\"custom-color-picker__title\\">Title</div><div class=\\"custom-color-picker__wrapper\\"><div class=\\"custom-color-picker__swatch\\"><div class=\\"custom-color-picker__color\\" style=\\"background:#000000\\"></div></div><div style=\\"position:relative\\"><input style=\\"border:none;outline:none;font-size:14px\\" value=\\"#000000\\" spellcheck=\\"false\\"/><span style=\\"display:none\\">hex</span></div></div></div><div class=\\"custom-color-picker\\"><div class=\\"custom-color-picker__title\\">Legend</div><div class=\\"custom-color-picker__wrapper\\"><div class=\\"custom-color-picker__swatch\\"><div class=\\"custom-color-picker__color\\" style=\\"background:#000000\\"></div></div><div style=\\"position:relative\\"><input style=\\"border:none;outline:none;font-size:14px\\" value=\\"#000000\\" spellcheck=\\"false\\"/><span style=\\"display:none\\">hex</span></div></div></div><div class=\\"custom-color-picker\\"><div class=\\"custom-color-picker__title\\">Label</div><div class=\\"custom-color-picker__wrapper\\"><div class=\\"custom-color-picker__swatch\\"><div class=\\"custom-color-picker__color\\" style=\\"background:#000000\\"></div></div><div style=\\"position:relative\\"><input style=\\"border:none;outline:none;font-size:14px\\" value=\\"#000000\\" spellcheck=\\"false\\"/><span style=\\"display:none\\">hex</span></div></div></div><div class=\\"custom-color-picker\\"><div class=\\"custom-color-picker__title\\">Chart text</div><div class=\\"custom-color-picker__wrapper\\"><div class=\\"custom-color-picker__swatch\\"><div class=\\"custom-color-picker__color\\" style=\\"background:#000000\\"></div></div><div style=\\"position:relative\\"><input style=\\"border:none;outline:none;font-size:14px\\" value=\\"#000000\\" spellcheck=\\"false\\"/><span style=\\"display:none\\">hex</span></div></div></div><div class=\\"custom-color-picker\\"><div class=\\"custom-color-picker__title\\">Axis text</div><div class=\\"custom-color-picker__wrapper\\"><div class=\\"custom-color-picker__swatch\\"><div class=\\"custom-color-picker__color\\" style=\\"background:#000000\\"></div></div><div style=\\"position:relative\\"><input style=\\"border:none;outline:none;font-size:14px\\" value=\\"#000000\\" spellcheck=\\"false\\"/><span style=\\"display:none\\">hex</span></div></div></div><div class=\\"custom-color-picker\\"><div class=\\"custom-color-picker__title\\">Tooltip text</div><div class=\\"custom-color-picker__wrapper\\"><div class=\\"custom-color-picker__swatch\\"><div class=\\"custom-color-picker__color\\" style=\\"background:#000000\\"></div></div><div style=\\"position:relative\\"><input style=\\"border:none;outline:none;font-size:14px\\" value=\\"#000000\\" spellcheck=\\"false\\"/><span style=\\"display:none\\">hex</span></div></div></div><div class=\\"custom-color-picker\\"><div class=\\"custom-color-picker__title\\">Grid line</div><div class=\\"custom-color-picker__wrapper\\"><div class=\\"custom-color-picker__swatch\\"><div class=\\"custom-color-picker__color\\" style=\\"background:#000000\\"></div></div><div style=\\"position:relative\\"><input style=\\"border:none;outline:none;font-size:14px\\" value=\\"#000000\\" spellcheck=\\"false\\"/><span style=\\"display:none\\">hex</span></div></div></div><h2 class=\\"builder__header\\">Chart colors palette</h2><label>Predefined color palette</label><div class=\\"predefined-palette css-2b097c-container\\"><div class=\\" css-yk16xz-control\\"><div class=\\" css-1hwfws3\\"><div class=\\" css-1uccc91-singleValue\\">Default</div><div class=\\"css-1g6gooi\\"><div class=\\"\\" style=\\"display:inline-block\\"><input type=\\"text\\" autoCapitalize=\\"none\\" autoComplete=\\"off\\" autoCorrect=\\"off\\" id=\\"react-select-4-input\\" spellcheck=\\"false\\" tabindex=\\"0\\" value=\\"\\" aria-autocomplete=\\"list\\" style=\\"box-sizing:content-box;width:1px;label:input;background:0;border:0;font-size:inherit;opacity:1;outline:0;padding:0;color:inherit\\"/><div style=\\"position:absolute;top:0;left:0;visibility:hidden;height:0;overflow:scroll;white-space:pre\\"></div></div></div></div><div class=\\" css-1wy0on6\\"><span class=\\" css-1okebmr-indicatorSeparator\\"></span><div aria-hidden=\\"true\\" class=\\" css-tlfecz-indicatorContainer\\"><svg height=\\"20\\" width=\\"20\\" viewBox=\\"0 0 20 20\\" aria-hidden=\\"true\\" focusable=\\"false\\" class=\\"css-19bqh2r\\"><path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\"></path></svg></div></div></div></div><div class=\\"color-palette__list\\"><button class=\\"color-palette__button btn-list btn\\" type=\\"button\\">Add</button></div></section><div class=\\"jsx-4097032988 builder__mockup\\"><h2 class=\\"jsx-4097032988 builder__header\\">Preview</h2><div class=\\"react-tabs\\" data-tabs=\\"true\\"><ul class=\\"react-tabs__tab-list\\" role=\\"tablist\\"><li class=\\"react-tabs__tab react-tabs__tab--selected\\" role=\\"tab\\" id=\\"react-tabs-0\\" aria-selected=\\"true\\" aria-disabled=\\"false\\" aria-controls=\\"react-tabs-1\\" tabindex=\\"0\\">Chart</li><li class=\\"react-tabs__tab\\" role=\\"tab\\" id=\\"react-tabs-2\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" aria-controls=\\"react-tabs-3\\">CSS</li><li class=\\"react-tabs__tab\\" role=\\"tab\\" id=\\"react-tabs-4\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" aria-controls=\\"react-tabs-5\\">Javascript</li></ul><div class=\\"react-tabs__tab-panel react-tabs__tab-panel--selected\\" role=\\"tabpanel\\" id=\\"react-tabs-1\\" aria-labelledby=\\"react-tabs-0\\"><div class=\\"builder__default-charts\\"><div class=\\"keen-dataviz\\"></div><div class=\\"keen-dataviz\\"></div><div class=\\"keen-dataviz\\"></div><div class=\\"keen-dataviz\\"></div></div></div><div class=\\"react-tabs__tab-panel\\" role=\\"tabpanel\\" id=\\"react-tabs-3\\" aria-labelledby=\\"react-tabs-2\\"></div><div class=\\"react-tabs__tab-panel\\" role=\\"tabpanel\\" id=\\"react-tabs-5\\" aria-labelledby=\\"react-tabs-4\\"></div></div></div></div>"`;
